modules/installing-gcp-user-defined-labels-and-tags.adoc

```diff
@@ -0,0 +1,123 @@
+
+User-defined tags are attached to resources created by {product-title} core operator `cluster-image-registry-operator` and not on the resources created by any other operators or the kubernetes in-tree components. In {product-title} version {product-version}, resources created by `installer`, `machine-api-provider-gcp`, `gcp-filestore-csi-driver-operator`, `gcp-pd-csi-driver-operator` are not tagged.
+
+[NOTE]
+====
+Labels & tags once defined during the install-config creation phase cannot be modified or updated with new labels & tags after cluster creation.
+====
+
+Support for user-defined labels & tags for GCP resources is available in {product-title} version {product-version} as Technology Preview (TP).
+
+Service Account used for cluster creation must have the `TagUser` roles assigned to it, required for tags defined at both organization and project level.
+
+You can use `.platform.gcp.userLabels` and `platform.gcp.userTags` fields to define the list of desired labels & tags in `install-config.yaml`. `.platform.gcp.userLabels` and `.platform.gcp.userTags` fields are a yaml array of objects type.
+
+.Sample `install-config.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+featureSet: TechPreviewNoUpgrade
+platform:
+ gcp:
+   userLabels:
+   - key: <label_key>
+     value: <label_value>
+   userTags:
+   - parentID: <OrganizationID/ProjectID>
+     key: <tag_key_short_name>
+     value: <tag_value_short_name>
+----
+
+The user-defined labels have the following requirements:
+
+* A label key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A label key and value must contain only lowercase letters, numeric characters, underscores, and dashes (i.e. _-).
+* A label key must start with a lowercase letter.
+* You can configure a maximum of 32 labels per resource.
+and each resource can have a maximum of 64 labels, and 32 labels are reserved for internal use by {product-title}.
+
+The user-defined tags have the following requirements:
+
+* Tag Keys and Tag Values must already exist and will not be created by {product-title}.
+* A tag `parentID` can be either `OrganizationID` or `ProjectID`.
+** An `OrganizationID` must consist of decimal numbers, and cannot have leading zeros.
+** A `ProjectID` must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.
+* A tag key and value must have a minimum of 1 character and can have a maximum of 63 characters.
+* A tag key must contain only uppercase and lowercase alphanumeric characters, hyphens, underscores, and periods (i.e. ._-).
```

Comment:
Suggest to order the special characters by the same order of the words, for example, if the words are "hyphens, underscores, and periods", the special characters shall be listed as "-_.", thanks!